src/utils/ProteomicsLFQ.cpp

```diff
@@ -0,0 +1,1598 @@
+        OPENMS_LOG_INFO << "after alignment:" << endl;
+        OPENMS_LOG_INFO << (int)s.percentiles_after[100] << "\t"
+                  << (int)s.percentiles_after[99] << "\t" 
+                  << (int)s.percentiles_after[95] << "\t" 
+                  << (int)s.percentiles_after[90] << "\t" 
+                  << (int)s.percentiles_after[50] << "\t" 
+                  << (int)s.percentiles_after[25] << endl; 
+      }
+
+      double max_alignment_diff = std::max_element(alignment_stats.begin(), alignment_stats.end(),
+              [](TrafoStat a, TrafoStat b) 
+              { return a.percentiles_after[100] > b.percentiles_after[100]; })->percentiles_after[100];
+      // sometimes, very good alignments might lead to bad overall performance. Choose 2 minutes as minimum.
+     OPENMS_LOG_INFO << "Max alignment difference (seconds): " << max_alignment_diff << endl;
+      max_alignment_diff = std::max(max_alignment_diff, 120.0);
+      return max_alignment_diff;
+    }
+    return 0;
+  }
+
+  void transform_(
+    vector<FeatureMap>& feature_maps, 
+    vector<TransformationDescription>& transformations
+  )
+  {
+    if (feature_maps.size() > 1 && !transformations.empty())
+    {
+      // Apply transformations
+      for (Size i = 0; i < feature_maps.size(); ++i)
+      {
+        try 
+        {
+          MapAlignmentTransformer::transformRetentionTimes(feature_maps[i],
+            transformations[i]);
+        } catch (Exception::IllegalArgument e)
+        {
+         OPENMS_LOG_WARN << e.getMessage() << endl;
+        }
+          
+        if (debug_level_ > 666)
+        {
+          // plot with e.g.:
+          // Rscript ../share/OpenMS/SCRIPTS/plot_trafo.R debug_trafo_1.trafoXML debug_trafo_1.pdf
+          TransformationXMLFile().store("debug_trafo_" + String(i) + ".trafoXML", transformations[i]);
+        }
+      }
+    }
+  }
+
+  //-------------------------------------------------------------
+  // Link all features of this fraction
+  //-------------------------------------------------------------
+  void link_(
+    vector<FeatureMap> & feature_maps, 
+    double median_fwhm,
+    double max_alignment_diff,
+    ConsensusMap & consensus_fraction
+  )
+  {
+    Param fl_param = getParam_().copy("Linking:", true);
+    writeDebug_("Parameters passed to feature grouping algorithm", fl_param, 3);
+
+    writeDebug_("Linking: " + String(feature_maps.size()) + " features.", 1);
+
+    // grouping tolerance = max alignment error + median FWHM
+    FeatureGroupingAlgorithmQT linker;
+    fl_param.setValue("distance_RT:max_difference", 2.0 * max_alignment_diff + 2.0 * median_fwhm);
+    fl_param.setValue("distance_MZ:max_difference", 10.0);
+    fl_param.setValue("distance_MZ:unit", "ppm");
+    fl_param.setValue("distance_MZ:weight", 5.0);
+    fl_param.setValue("distance_intensity:weight", 0.1); 
+    fl_param.setValue("use_identifications", "true"); 
+/*
+    FeatureGroupingAlgorithmKD linker;
```

Comment:
can the user select between FeatureGroupingAlgorithmQT and FeatureGroupingAlgorithmKD ?

Reply:
No, I would prefer choosing one that works for all kind of studies. The code is just there to quickly check how the other one performs.